File: 0735-asteroid-collision/0735-asteroid-collision.cs
public class Solution {
    public int[] AsteroidCollision(int[] A) {
        Stack<int> st = new Stack<int>();
        foreach(int a in A) {
            int flag = 1;
            while(st.Count > 0 && (st.Peek() > 0 && a < 0)) {
                if(Math.Abs(a) > Math.Abs(st.Peek())) {
                    st.Pop(); continue;
                }
                if(Math.Abs(a) == Math.Abs(st.Peek())) {
                    st.Pop();
                }
                flag = 0;
                break;
            }
            if(flag == 1) st.Push(a);
        }
        int[] ans = new int[st.Count];
        for(int i = st.Count - 1; i >= 0; --i) {
            ans[i] = st.Peek(); st.Pop();
        }
        return ans;

    }
}

File: 0735-asteroid-collision/0735-asteroid-collision.ts
function asteroidCollision(A: number[]): number[] {
    let st: number[] = [];
    for (let a of A) {
        let flag: number = 1;
        while (st.length > 0 && (st[st.length - 1] > 0 && a < 0)) {
            if (Math.abs(a) > Math.abs(st[st.length - 1])) {
                st.pop();
                continue;
            }
            if (Math.abs(a) === Math.abs(st[st.length - 1])) {
                st.pop();
            }
            flag = 0;
            break;
        }
        if (flag === 1) {
            st.push(a);
        }
    }
    let ans: number[] = new Array(st.length);
    for (let i: number = st.length - 1; i >= 0; --i) {
        ans[i] = st[st.length - 1];
        st.pop();
    }
    return ans;
};